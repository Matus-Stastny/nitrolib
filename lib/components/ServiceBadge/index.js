"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _Badge = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Badge"));

var _StarFull = _interopRequireDefault(require("@kiwicom/orbit-components/lib/icons/StarFull"));

var _Plus = _interopRequireDefault(require("@kiwicom/orbit-components/lib/icons/Plus"));

var _Text = _interopRequireDefault(require("../Text"));

var _ref2 =
/*#__PURE__*/
React.createElement(_StarFull.default, {
  customColor: "#f9971e"
});

var _ref3 =
/*#__PURE__*/
React.createElement(_Plus.default, {
  customColor: "#0176d2"
});

var _ref4 =
/*#__PURE__*/
React.createElement(_Text.default, {
  t: "booking.service_packages.premium",
  type: "white",
  size: "small"
});

var _ref5 =
/*#__PURE__*/
React.createElement(_Text.default, {
  t: "booking.service_packages.plus",
  size: "small"
});

var ServiceBadge = function ServiceBadge(_ref) {
  var premium = _ref.premium;
  return React.createElement(_Badge.default, {
    type: premium ? "dark" : "neutral",
    icon: premium ? _ref2 : _ref3
  }, premium ? _ref4 : _ref5);
};

var _default = ServiceBadge;
exports.default = _default;