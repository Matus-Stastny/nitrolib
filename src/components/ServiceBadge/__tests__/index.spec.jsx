// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import ServiceBadge from "..";

describe("#ServiceBadge", () => {
  test("render", () => {
    const wrapper = shallow(<ServiceBadge />);

    expect(wrapper).toMatchSnapshot();
  });

  test("translation plus", () => {
    const wrapperPlus = shallow(<ServiceBadge />);
    expect(wrapperPlus.find("Text").prop("t")).toBe("booking.service_packages.plus");
  });

  test("translation premium", () => {
    const wrapperPlus = shallow(<ServiceBadge premium />);
    expect(wrapperPlus.find("Text").prop("t")).toBe("booking.service_packages.premium");
  });
});
