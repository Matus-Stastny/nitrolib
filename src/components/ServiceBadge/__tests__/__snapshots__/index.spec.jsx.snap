// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ServiceBadge render 1`] = `
<Badge
  icon={
    <Plus
      customColor="#0176d2"
    />
  }
  type="neutral"
>
  <Text
    size="small"
    t="booking.service_packages.plus"
  />
</Badge>
`;
